### PR TITLE
Change to force a rebuild of the pollset on flag changes

### PR DESCRIPTION
**Pull request by guillaumeautran**
This change fixes issues seen in #1357
Goes with #1281

### CODE REVIEW

File: clients/roscpp/src/libros/io.cpp [hunk 1]
```diff
@@ -119,6 +119,8 @@ void close_socket_watcher(int fd) {
 void add_socket_to_watcher(int epfd, int fd) {
 #if defined(HAVE_EPOLL)
   struct epoll_event ev;
+  bzero(&ev, sizeof(ev));
```
Comment (guillaumeautran) [inline clients/roscpp/src/libros/io.cpp:122]: Fixing valgrind warnings.

File: clients/roscpp/src/libros/io.cpp [hunk 2]
```diff
@@ -147,6 +149,8 @@ void del_socket_from_watcher(int epfd, int fd) {
 void set_events_on_socket(int epfd, int fd, int events) {
 #if defined(HAVE_EPOLL)
   struct epoll_event ev;
+  bzero(&ev, sizeof(ev));
```
Comment (guillaumeautran) [inline clients/roscpp/src/libros/io.cpp:152]: Fixing valgrind warnings.

File: test/test_roscpp/test/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,6 @@
 catkin_add_gtest(${PROJECT_NAME}-test_version test_version.cpp)
 if(TARGET ${PROJECT_NAME}-test_version)
-  target_link_libraries(${PROJECT_NAME}-test_version)
+  target_link_libraries(${PROJECT_NAME}-test_version ${catkin_LIBRARIES})
```
Comment (guillaumeautran) [inline test/test_roscpp/test/CMakeLists.txt:3]: I could not build and run the unit tests without the following changes.

Comment (mikepurvis) [reply test/test_roscpp/test/CMakeLists.txt:3]: Did you end up just running the tests on Linux but with a build where `HAVE_EPOLL` was forced to 0?

Comment (guillaumeautran) [reply test/test_roscpp/test/CMakeLists.txt:3]: That's correct yes.

Comment (mikepurvis) [reply test/test_roscpp/test/CMakeLists.txt:3]: And prior to this fix the tests failed to complete successfully?

Comment (guillaumeautran) [reply test/test_roscpp/test/CMakeLists.txt:3]: Yes, sorry, I should have mentioned that. 

Also, I am looking at the current build failure.

Comment (mikepurvis) [reply test/test_roscpp/test/CMakeLists.txt:3]: My impression is that that test has been flakey for a while— it definitely has a timing sensitivity and does _sometimes_ pass on the buildfarm. However, if you have further insights on it, that would be great.

**Comment by guillaumeautran** [comment]
I cannot figure out why the test fails on the build farm. They obviously pass just fine on my laptop which makes it hard to troubleshoot...

**Comment by mikepurvis** [comment]
LGTM, and has the thumbs-up from @mpflanzer in https://github.com/ros/ros_comm/issues/1357#issuecomment-388000450.

IMO this should merge now and we can treat the test regression as separate since it predates this change.

**Pull request merged**